app-libs/exchange-oracle/src/coingecko.rs

```diff
@@ -0,0 +1,176 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+use crate::{
+	error::{Error, Error::RestClientError},
+	ExchangeRateGet,
+};
+use itc_rest_client::{http_client::HttpClient, rest_client::RestClient, RestGet, RestPath};
+use log::*;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use url::Url;
+
+static COINGECKO_URL: &str = "https://api.coingecko.com";
+static COINGECKO_PARAM_CURRENCY: &str = "vs_currency";
+static COINGECKO_PARAM_COIN: &str = "ids";
+static COINGECKO_PATH: &str = "api/v3/coins/markets";
```

Comment:
I think in cases of these strings, it is better to use `const` instead of `static`, to have inline replacements rather than global variables.